comp120_1507866.tex

```diff
@@ -0,0 +1,54 @@
+
+\begin{document}
+	
+\maketitle
+	
+\abstract{This report will look over the last semester and reflect on what I believe to have been my three main weaknesses}.
+	
+\section{Introduction}
+After reviewing my weekly reports for this semester I feel that my three main weaknesses are my time management, problems solving and critiquing skills.
+
+\section{Time Management}
+Time management is the skill that needs the most improvement. I have found when I have many assignments set it can be overwhelming. It would be better to plan and assign time to each one.
+
+Currently I assign time to an assignment and either not do enough work or focus too much on other assignments. This is a problem as it means I am not spending enough time on some assignments.
+
+Previously I have been using Trello boards to plan my Kivy app. In that particular assignment I found Trello helpful. I think a possible way to improve my time management would be to use Trello boards for more assignments and make them more detailed. The use of Trello boards also links to using sprint plans. For future assignments I am aiming to use more sprint plans. I can plan what needs to be done in an assignment and then use a sprint plan to stay on track. 
+
+
+\section{Problem Solving}
+Another weakness from the last semester is my problem solving abilities. This was initially a problem for the SpaceChem worksheet. On many of the levels I found that my solutions were poor in both the number of elapsed cycles and symbols used.
+
+Problems solving also effects my coding abilities. For both the university course and afterwards, improving on these skills should help me find better solutions when programming.
+
+One way I intend to improve these skills is to complete SpaceChem and other problem solving games. I found previously when playing SpaceChem that completing more levels gave me ideas to improve on previous ones. Completing such games should help improve my problem solving.  This will be measurable in the games feedback on each level. Another action is to read the Think Like A Programmer textbook which addresses problem solving.
+I will complete these actions within the next two months so I can improve for the next assignments.
+
+\section{Critiquing Code}
+A further weakness is my critical thinking. Currently I find it hard to critique other people's code and often find that I cannot sufficiently explain the issue. I also find it hard to analyse my own code. Improving on this skill will help me improve my code. It will also help me to give my classmates more constructive feedback in code reviews.
+I intend to improve on this by reviewing other people's code on GitHub more often than just the in class review sessions.  This will help me practice the skill and hopefully improve on it. I will also practice with my own code. I intend to do this by structuring it better and testing more frequently. This will prevent some issues from arising and hopefully make them simpler to solve if they do.
```

Comment:
Critiquing code could be more analytical instead of just saying you found it hard maybe explain why and how the resulting issue of not being able to give constructive feedback made you feel as well as talking about if this skill has developed over the semester.